splitio/sync/segment.py

```diff
@@ -174,13 +176,124 @@ def synchronize_segments(self, segment_names = None, dont_wait = False):
         """
         if segment_names is None:
             segment_names = self._split_storage.get_segment_names()
-            
+
         for segment_name in segment_names:
             self._worker_pool.submit_work(segment_name)
         if (dont_wait):
             return True
         return not self._worker_pool.wait_for_completion()
-    
+
+    def segment_exist_in_storage(self, segment_name):
+        """
+        Check if a segment exists in the storage
+
+        :param segment_name: Name of the segment
+        :type segment_name: str
+
+        :return: True if segment exist. False otherwise.
+        :rtype: bool
+        """
+        return self._segment_storage.get(segment_name) != None
+
+class LocalSegmentSynchronizer(object):
+    def __init__(self, segment_folder, split_storage, segment_storage):
+        """
+        Class constructor.
+
+        :param segment_api: API to retrieve segments from backend.
+        :type segment_api: splitio.api.SegmentApi
+
+        :param split_storage: Split Storage.
+        :type split_storage: splitio.storage.InMemorySplitStorage
+
+        :param segment_storage: Segment storage reference.
+        :type segment_storage: splitio.storage.SegmentStorage
+
+        """
+        self._segment_folder = segment_folder
+        self._split_storage = split_storage
+        self._segment_storage = segment_storage
+
+    def synchronize_segment(self, segment_name, till=None):
+        """
+        Update a segment from queue
+
+        :param segment_name: Name of the segment to update.
+        :type segment_name: str
+
+        :param till: ChangeNumber received.
+        :type till: int
+
+        :return: True if no error occurs. False otherwise.
+        :rtype: bool
+        """
+        try:
+            fetched = self._read_segment_from_json_file(segment_name)
+            if not self.segment_exist_in_storage(segment_name):
+                    self._segment_storage.put(segments.from_raw(fetched))
+                    self._segment_storage.set_change_number(segment_name, self._get_sha(json.dumps(fetched)))
+                    _LOGGER.debug("segment %s is added to storage", segment_name)
+            else:
+                if self._segment_storage.get_change_number(segment_name) != self._get_sha(json.dumps(fetched)):
```

Comment:
are you comparing changeNumber against sha? you can track last sha in a dict in this class

Reply:
changenumber is not a number from sha, is the last time the definition was updated

Reply:
corrected to use only change number